scene/gui/texture_rect.cpp

```diff
@@ -95,6 +95,15 @@ void TextureRect::_notification(int p_what) {
 			} break;
 		}
 
+		Ref<AtlasTexture> p_atlas = texture;
+
+		if (p_atlas.is_valid() && region.has_no_area()) {
+			Size2 scale_size(size.width / texture->get_width(), size.height / texture->get_height());
+
+			offset.width += hflip ? p_atlas->get_margin().get_position().width * size.width / texture->get_width() * 2 : 0;
```

Comment:
Didn't you mean to use `scale_size.width` here?

Reply:
Fixed.